app/assets/javascripts/models/page/project_page.js.coffee

```diff
@@ -45,6 +49,7 @@ ns.ProjectPage = do (ko,
       @ui =
         temporal: new TemporalModel(@query)
         projectList: projectList
+        projectPage: this
```

Comment:
This seems redundant -- is the reason for this the code below `$root.ui.projectPage.hasType()`?  If so, can that just be `$root.hasType()`?

Reply:
You have it right - setting 'projectPage' to equal itself exposes it to the view - without that line, the call to $root.ui.projectPage.hasType will fail. 

You're also right in that $root.hasType() does work!  However, isn't using the full path more descriptive?  I'm a little torn - could go either way.

Reply:
Ahhhhh - I see what you mean.  If you use $root.hasType, then the assignment to @ui isn't necessary.

I'll go with that - nice catch.  :)